### PR TITLE
LSR-931 Expose User object in SessionContext

### PR DESCRIPTION
**Pull request by victorg1991**

### CODE REVIEW

**Comment by dgarciasarai** [comment]
👍  LGTM

**Comment by nhpatt** [comment]
👍

File: ios/Framework/Core/Auth/LoginScreenlet/LoginScreenlet.swift [hunk 1]
```diff
@@ -72,7 +72,7 @@ public class LoginScreenlet: BaseScreenlet, BasicAuthBasedType {
 			viewModel.userName = SessionContext.currentContext?.basicAuthUsername
 			viewModel.password = SessionContext.currentContext?.basicAuthPassword
 
-			let userAttributes = SessionContext.currentContext!.userAttributes
```
Comment (jmnavarro) [inline ios/Framework/Core/Auth/LoginScreenlet/LoginScreenlet.swift]: Add a computed property called `userAttributes` to maintain backwards compatibility

File: ios/Framework/Core/Auth/SignUpScreenlet/Connectors/UpdateCurrentUserLiferayConnector.swift [hunk 2]
```diff
@@ -48,11 +48,11 @@ public class UpdateCurrentUserLiferayConnector: ServerConnector {
 	}
 
 	public func attributeAsString(key: String) -> String {
-		return SessionContext.currentContext?.userAttribute(key) as! String
```
Comment (jmnavarro) [inline ios/Framework/Core/Auth/SignUpScreenlet/Connectors/UpdateCurrentUserLiferayConnector.swift]: Same here: add `userAttribute(key)` method.
Keep in mind the [Liskov Substitution Principle](https://en.wikipedia.org/wiki/Liskov_substitution_principle), or as a simple rule of thumb: don't chain more than one period in an expression (don't access one object internals, just ask the object to do something)

File: ios/Framework/Core/Auth/UserPortraitScreenlet/UserPortraitScreenlet.swift [hunk 3]
```diff
@@ -79,7 +79,7 @@ public class UserPortraitScreenlet: BaseScreenlet {
 	}
 
 	public func loadLoggedUserPortrait() -> Bool {
-		guard let userId = SessionContext.currentContext?.userId else {
```
Comment (jmnavarro) [inline ios/Framework/Core/Auth/UserPortraitScreenlet/UserPortraitScreenlet.swift]: Same here

File: ios/Framework/Core/Models/User.swift [hunk 4]
```diff
@@ -54,11 +50,27 @@ import Foundation
 		return int64Value("userId") ?? 0
 	}
 
+	public init(attributes: [String : AnyObject]) {
+		self.attributes = attributes
+
+		super.init()
+	}
+
 	private func int64Value(key: String) -> Int64? {
```
Comment (jmnavarro) [inline ios/Framework/Core/Models/User.swift]: Move these helper methods to somewhere?

File: ios/Framework/Core/Auth/LoginScreenlet/LoginScreenlet.swift (hunk 1 above)
Comment (victorg1991) [inline ios/Framework/Core/Auth/LoginScreenlet/LoginScreenlet.swift]: 👍 in c5caab8

File: ios/Framework/Core/Auth/UserPortraitScreenlet/UserPortraitScreenlet.swift (hunk 3 above)
Comment (victorg1991) [inline ios/Framework/Core/Auth/UserPortraitScreenlet/UserPortraitScreenlet.swift]: 👍 in c5caab8

File: ios/Framework/Core/Auth/SignUpScreenlet/Connectors/UpdateCurrentUserLiferayConnector.swift (hunk 2 above)
Comment (victorg1991) [inline ios/Framework/Core/Auth/SignUpScreenlet/Connectors/UpdateCurrentUserLiferayConnector.swift]: 👍 2799c61

**Pull request merged**